### PR TITLE
[ICDS Dashboard] Performance improvement of THR report

### PR DESCRIPTION
**Pull request by Rohit25negi**
Adding these columns to AWC table so that we dont need to make join between tables which slows down the query when queried on level 5.
Ticket: https://dimagi-dev.atlassian.net/browse/ICDS-932

NOTE: Feel free to highlight the nits as well. Would love to raise code quality.

### CODE REVIEW

File: custom/icds_reports/migrations/sql_templates/database_views/thr_report_view.sql
Comment (Rohit25negi) [inline custom/icds_reports/migrations/sql_templates/database_views/thr_report_view.sql]: It is not making sense to me to make different to deploy this view change and wait for 1 week this change to reflect.
for this, I can do that data updation in the agg table on the night of deploy before people wakes up next day

**Comment by Rohit25negi** [comment]
@calellowitz  can you review this PR.

**Review by calellowitz — APPROVED** [review]

**Comment by emord** [comment]
Is there any update on this work? I've noticed these thr_report_monthly queries taking a lot of disk IO recently and this seems like a PR that would help

**Comment by Rohit25negi** [comment]
@emord  I have fixed the merged conflicts and this is good to go if the change looks good to you as reviewer.
PS: THIs need some data filling right after deploying. So please ping in advance if you are going to merge it adhocly. I am expecting this will get deployed on mOnday.

**Review by emord — APPROVED** [review]
Approval is conditional on first point

> THIs need some data filling right after deploying. 

Thanks for pointing this out. Is it possible to do it before the deploy? It appears that the columns already exist in `agg_awc` meaning that the data filling could happen before the deploy so that this is not dependent. (Provided that the old agg query does not run between data filling and deploy since that would overwrite the data)

> So please ping in advance if you are going to merge it adhocly

This is a good use of the "open for review; dont merge" label which I just put on this PR

**Comment by Rohit25negi** [comment]
> Is it possible to do it before the deploy?

yes, Data can be filled before deploy. I will do the refill Today. Once the script is ready I will merge this and do the data fill.

@emord  @calellowitz

**Comment by Rohit25negi** [comment]
This data filling query is approved by Cal on Slack. But I missed today's deploy. But, I will refill the data till aug. And let the setp and Oct be filled when the code is deployed.

@calellowitz  @emord  fyi

**Comment by emord** [comment]
> This data filling query is approved by Cal on Slack

Should this sort of query/command be sent out for review? Even if it isn't merged into the codebase, it could serve as a better check than a slack message, a place to point to when showing others how to do similar queries and the burden of approval doesn't necessarily fall on one person to approve

**Comment by Rohit25negi** [comment]
> > This data filling query is approved by Cal on Slack
> 
> Should this sort of query/command be sent out for review? Even if it isn't merged into the codebase, it could serve as a better check than a slack message, a place to point to when showing others how to do similar queries and the burden of approval doesn't necessarily fall on one person to approve

you mean review on github? Should It mean raising another Pr with Just the query in it but that Pr will never get merged?

**Comment by Rohit25negi** [comment]
Though its already reviewed by Cal. But putting the script [here](https://github.com/dimagi/commcare-hq/pull/25695) for future reference. 
@emord fyi. Were you imagining this in different way.

**Comment by Rohit25negi** [comment]
Merging this.

**Pull request merged**
